tests/integration/integration_suite_test.go

```diff
@@ -345,17 +353,46 @@ func testStargate(t *testing.T, namespace string) {
 }
 
 func TestUpgradeScenario(t *testing.T) {
-	namespace := initializeCluster(t)
-	// Install first production version
-	InstallK8ssandraFromRepo(t, namespace, "1.0.0")
-	checkResourcePresenceForReaper(t, namespace)
-	waitForReaperPod(t, namespace)
+	for _, startVersion := range upgradeStartVersions {
+		namespace := initializeCluster(t)
+		// Install first production version
+		DeployClusterWithValues(t, namespace, "default", "cluster_with_reaper.yaml", 1, false, false, startVersion)
+		checkResourcePresenceForReaper(t, namespace)
+		waitForReaperPod(t, namespace)
+
+		// Upgrade to current version
+		DeployClusterWithValues(t, namespace, "default", "cluster_with_reaper.yaml", 1, true, true, "")
+		checkResourcePresenceForReaper(t, namespace)
+		waitForReaperPod(t, namespace)
+		checkReaperRegistered(t, namespace)
+
+		cleanupCluster(t, namespace, t.Failed())
```

Comment:
If this fails, we shouldn't ever continue (as it will fail to same name etc issue). Should probably just do if t.Failed() .. { t.FailNow() } or equal.